src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -55,9 +58,25 @@ protected QueryResultRetrieverBase(DocumentDatabase database, IndexQueryServerSi
             FieldsToFetch = fieldsToFetch;
 
             _blittableTraverser = reduceResults ? BlittableJsonTraverser.FlatMapReduceResults : BlittableJsonTraverser.Default;
+
+        }
+
+        protected void FinishDocumentSetup(Document doc, float score, int index)
+        {
+            if (doc == null)
+                return;
+
+            doc.IndexScore = score;
+            if (_query.Distances != null)
+            {
+                var distanceInRadians = _query.Distances.Get(index);
+
+                doc.Distance = distanceInRadians * DistanceUtils.RADIANS_TO_DEGREES * DistanceUtils.MILES_TO_KM;
```

Comment:
is this valid?

Reply:
BTW, the comparator has a spatial field with option where you have access to units. Maybe we can take leverage from that?

Reply:
No, it isn't. This was meant just to get me working on _something_ so we'll have a result.